### DISCUSSION TITLE
How can i properly execute encode(hmac( '89409', 'Klucz123', 'md5'), 'hex'); on cipher calc ?

### DISCUSSION THREAD

**Discussion by kasiahayder** [wierkstudio/ciphereditor#27] [Help] [answered]
How can i properly execute encode(hmac( '89409', 'Klucz123', 'md5'), 'hex'); on cipher calc ? Now i obtain different output than in other calculators :/

**Comment by ffraenz** [comment]
You should be able to use the [HMAC function operation](https://ciphereditor.com/explore/hmac) and enter `89409` as the message, `Klucz123`as the key, and select `SHA-1` as the algorithm. As the message is entered in text, the hash is returned as hex text as well. Does this work for you?

**Comment by kasiahayder** [comment]
![ZRZUT 10229 2023-03-31 13 17](https://user-images.githubusercontent.com/14802273/229106271-5938ec06-ff59-4b0b-ae5b-4cc92cf31030.jpg)
![ZRZUT 10228 2023-03-31 13 17](https://user-images.githubusercontent.com/14802273/229106282-7aea8bbe-f9fc-4701-848a-18a91b35a3de.jpg)
https://www.freeformatter.com/hmac-generator.html#before-output versus https://cryptii.com/pipes/hmac.
Please, compare yourself; I am not sure if am doing sthng wrong.
Thank you

**Reply by ffraenz** [reply]
Please note that cryptii reports a validation error for the key field (in red) as `Klucz123` is not a hex string (expected by cryptii). Here, you need to first turn `Klucz123` into the hex string `4b 6c 75 63 7a 31 32 33`, resulting in the HMAC-SHA-1 hash `985dc8e03e99317d66907f6fc825062bbb5bcb85`.

ciphereditor returns the same result as cryptii and the third party tool you mentioned.

**Comment by kasiahayder** [comment, marked as answer]
Thank you very much for your help

pt., 31 mar 2023, 14:44 użytkownik Fränz Friederes ***@***.***>
napisał:
